src/metabase/pulse/parameters.clj

```diff
@@ -49,7 +49,10 @@
   "Given a dashcard and the parameters on a dashboard, returns the dashcard with any parameter values appropriately
   substituted into connected variables in the text."
   [dashcard parameters]
-  (let [text               (-> dashcard :visualization_settings :text)
+  (let [dashcard           (if (= "heading" (get-in dashcard [:visualization_settings :virtual_card :display]))
+                             (update-in dashcard [:visualization_settings :text] #(str "## " (shared.params/escape-chars % shared.params/escaped-chars-regex)))
```

Comment:
Approach for escaping looks good, but I think it might be a bit nicer to lift up this logic into `metabase.pulse`, maybe as a helper function that's called from `dashcard->part`. This namespace is specifically about parameter substitution which isn't relevant here